python/core/auto_generated/vector/qgsvectorlayer.sip.in

```diff
@@ -1082,40 +1082,6 @@ Determines if this vector layer has features.
 .. versionadded:: 3.4
 %End
 
- void setDataSource( const QString &dataSource, const QString &baseName, const QString &provider, bool loadDefaultStyleFlag = false ) /Deprecated/;
```

Comment:
This one is also an API break, no?

Reply:
It is not an API break because the method is provide by QgsMapLayer, so the method for QgsVectorLayer instances is always available.

If I want to unvirtualize QgsMapServer::setDataSource methods, I have to remove the specific QgsVectorLayer::setDataSource. And to avoid an other API break than unvirtualize, I move the method to QgsMapLayer.